homeassistant/components/weather/__init__.py

```diff
@@ -153,13 +404,40 @@ def humidity(self) -> float | None:
 
     @property
     def wind_speed(self) -> float | None:
-        """Return the wind speed in native units."""
+        """Return the wind_speed for backward compatibility.
+
+        Should not be set by integrations.
+        """
         return self._attr_wind_speed
 
     @property
-    def wind_speed_unit(self) -> str | None:
+    def native_wind_speed(self) -> float | None:
+        """Return the wind speed in native units."""
+        return self._attr_native_wind_speed
+
+    @property
+    def native_wind_speed_unit(self) -> str | None:
         """Return the native unit of measurement for wind speed."""
-        return self._attr_wind_speed_unit
+        return self._attr_native_wind_speed_unit
+
+    @property
+    def wind_speed_unit(self) -> str:
```

Comment:
Do like this for backwards compatibility:
```suggestion
        """Return the wind_speed unit for backward compatibility.
        Should not be set by integrations.
        """
        return self._attr_wind_speed_unit

    @property
    def _wind_speed_unit(self) -> str:
```